### PR TITLE
Scala 2.12.8 update

### PR DESCRIPTION
**Pull request by cquiroz**
It would be nice upgrading to 2.12.7 to get better compilation speeds. This PR fixes almost evrything but there is a regression:

https://github.com/scala/bug/issues/11174

### CODE REVIEW

**Comment by tpolecat** [comment]
Could we leave the Scala version at 2.12.6 for the client code and upgrade everything else? I talked to Seth and they're not sure what the plan is re: 11174.

**Comment by cquiroz** [comment]
Hmm, i didn't think of that. let me try. of course it is a pity since that's the place where I want a faster compiler

**Comment by cquiroz** [comment]
Somehow I can’t do this, I set the scala version for client but the compilation error persist

Gotta love sbt

**Comment by cquiroz** [comment]
We can proceed with this PR now that scala 2.12.8 has been released.

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala [hunk 1]
```diff
@@ -38,12 +38,15 @@ class ObservationsProgressStateHandler[M](
...
+          if (curStep.observeStatus === ActionStatus.Completed && !curStep.isObservePaused) {
+            updatedL(
+              AllObservationsProgressState
+                .progressByIdL(e.obsId, curSIdx)
+                .set(none))
+          } else {
+            noChange
+          }
 
       upd.getOrElse(noChange)
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala]: Same comment. Both codes seem equivalent, and IMO the old one looked better.

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/StepsTable.scala [hunk 2]
```diff
@@ -429,71 +429,60 @@ object StepsTable {
...
+      .map(i =>
+        Column(
+          Column.propsNoFlex(
+            ColWidths.ExposureWidth,
+            "exposure",
+            label = "Exposure",
+            className = SeqexecStyles.centeredCell.htmlClass,
+            cellRenderer = stepExposureRenderer(i.instrument)
+          )))
+      .filter(_ => exposureVisible)
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/StepsTable.scala]: Good catch

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala]: I agreee, the new compiler emits a warning about unused variables in the old code. This seems a bug in the compiler and this is the workaround

[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala]: I’ll ask about this in the scala channels though

**Comment by cquiroz** [comment]
In case you wonder why the update, scala 2.12.7/8 brings a compilation speed improvement of about 10%

**Comment by sraaphorst** [comment]
@cquiroz Excellent. Thanks for taking the time to answer my questions. I'm still confused, though, by what's happening with the `.map{ implicit S =>`... anyone explain? Why is `implicit` being used here? What is implicit?

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/handlers/ObservationsProgressStateHandler.scala]: This is a known issue in scala 
https://github.com/scala/bug/issues/11175

File: modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala [hunk 3]
```diff
@@ -202,7 +202,7 @@ object WebServerLauncher extends StreamApp[IO] with LogInitialization with Seqex
     // I have taken this from the examples at:
     // https://github.com/gvolpe/advanced-http4s/blob/master/src/main/scala/com/github/gvolpe/fs2/PubSub.scala
     // It's not very clear why we need to run this inside a Scheduler
-    Scheduler[IO](corePoolSize = 4).flatMap { implicit S =>
```
[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala]: Good question. I think I took this from an example while doing the painful scalaz-stream to fs2 port. I don't honestly know 😢

[the remark this answers is not in the document]
Comment (cquiroz) [reply modules/seqexec/web/server/src/main/scala/seqexec/web/server/http4s/WebServerLauncher.scala]: @tpolecat maybe knows?

**Comment by cquiroz** [comment]
Keeping this branch up to date is kind of painful so i’d Rather merge now

**Pull request merged**